File: ShareableJ/test/org/shareabledata/test/common/TimeAndMemoryLogger.java
package org.shareabledata.test.common;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import com.opencsv.*;

public class TimeAndMemoryLogger {
	private long initialTime = 0L;
	private String testCase;

	private long totalMemory = 0L;
	private long freeMemory = 0L;
	
	private ArrayList<String[]> contents;
	
	public TimeAndMemoryLogger() {
		contents = new ArrayList<String[]>();
		contents.add(new String[] {"Test Case ID", "CaseCounter", "TimeDifference", "TotalMemory", "FreeMemory"});
	}
	
	public void resetCounters() {
		this.initialTime = 0L;
	
		this.totalMemory = 0L;
		this.freeMemory = 0L;
	}
	
	public void setTestCaseName(String testCaseName) {
		this.testCase = testCaseName;
	}
	
	public void setInitialTimeAndMemory() {
		this.initialTime = System.nanoTime();
		this.totalMemory = Runtime.getRuntime().totalMemory();
		this.freeMemory = Runtime.getRuntime().freeMemory();
		contents.add(new String[] {testCase, ""+0,
									0+"",
									totalMemory+"",
									freeMemory+""}
					);
		
						
	}
	
	public void logTimeAndMemoryUsage(int caseIdNumber) {
		long endTime = System.nanoTime();
		long endTotalMemory = Runtime.getRuntime().totalMemory();
		long endFreeMemory = Runtime.getRuntime().freeMemory();
		
		
		contents.add(new String[] {testCase, ""+caseIdNumber,
									(endTime-this.initialTime)+"",
									endTotalMemory+"",
									endFreeMemory+""}
					);
			
	}
	
	public void writeToCSV(String FileName) throws IOException {
		String help = System.getProperty("user.dir");
		help = help + "\\execution\\";
		System.out.println(help);
		
		File outputFile = new File(help+FileName);
		FileWriter outputFileWriter = new FileWriter(outputFile);
		CSVWriter outputCSVWriter = new CSVWriter(outputFileWriter);
		outputCSVWriter.writeAll(contents);
		outputCSVWriter.close();
		
		
	}
	
	
	
		
}

File: Shareable/ShareableDATests/TimeAndMemoryLogger.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic.Devices;
using Microsoft.VisualBasic.FileIO;

namespace ShareableDATests
{
    public class TimeAndMemoryLogger
    {
        private long initialTime = 0L;
        private String testCase;

        private long totalMemory = 0L;
        private long freeMemory = 0L;

        StringBuilder contents;

        private const char delimiter = ',';
        public TimeAndMemoryLogger() {
            contents = new StringBuilder();
            contents.AppendLine("Test Case ID" + delimiter + "CaseCounter" + delimiter + "TimeDifference" + delimiter + "TotalMemory" + delimiter + "FreeMemory" );
        }

        public void resetCounters()
        {
            this.initialTime = 0L;

            this.totalMemory = 0L;
            this.freeMemory = 0L;
        }

        public void setTestCaseName(String testCaseName)
        {
            this.testCase = testCaseName;
        }


        public void setInitialTimeAndMemory()
        {

            ComputerInfo CI = new ComputerInfo();


            this.initialTime = TimeAndMemoryLogger.nanoTime();
            this.totalMemory = (long)CI.TotalPhysicalMemory; //I might be loosing presicion here?


            this.freeMemory = (long) CI.AvailablePhysicalMemory;
            contents.AppendLine(testCase+delimiter+
                            0+delimiter+
                            0+delimiter+
                            totalMemory+delimiter+
                            freeMemory);


        }

        public void logTimeAndMemoryUsage(int caseIdNumber)
        {
            ComputerInfo CI = new ComputerInfo();
            long endTime = TimeAndMemoryLogger.nanoTime();
            long endTotalMemory = (long)CI.TotalPhysicalMemory;
            long endFreeMemory = (long)CI.AvailablePhysicalMemory;


            contents.AppendLine(testCase+delimiter+
                            caseIdNumber+delimiter+
                            (endTime-this.initialTime)+delimiter+
                            endTotalMemory+delimiter+
                            endFreeMemory+delimiter);

        }

        public void writeToCSV(String FileName) 
        {
            string path = AppDomain.CurrentDomain.BaseDirectory;
            //Directory.GetCurrentDirectory(); points to VS execution
            path = Path.Combine(path, "execution", FileName);

            
            Console.WriteLine(path);

            StreamWriter Writer = new StreamWriter(path);



            Writer.Write(contents.ToString());
            Writer.Close();
        }

        private static long nanoTime()
        {
            long nano = 10000L * System.Diagnostics.Stopwatch.GetTimestamp();
            nano /= TimeSpan.TicksPerMillisecond;
            nano *= 100L;
            return nano;
        }





    }



    
}
